frog/responsive-images.rkt

```diff
@@ -0,0 +1,117 @@
+#lang racket/base
+
+(require net/url
+         racket/contract/base
+         racket/contract/region
+         racket/file
+         racket/function
+         racket/list
+         racket/path
+         racket/port
+         racket/system
+         racket/string
+         "params.rkt"
+         "util.rkt"
+         "verbosity.rkt"
+         (only-in srfi/1 list-index))
+
+(provide make-responsive clean-resized-images magick-available?)
+
+(define magick-available? (find-executable-path "magick"))
+
+(module+ test
+  (require rackunit))
+
+(define (image-width path)
+  (define (magick-width path)
+    (string->number
+     (with-output-to-string
+       (λ ()
+         ; Fails when running from inside DrRacket on Mac OS X due to pathologic PATH
+         (system (string-append "identify -format %w \"" (path->string path) "\""))))))
+  (define get-width magick-width)
+  (get-width path))
+
+(module+ test
+  (when magick-available?
+    (check-eq? (image-width (string->path "../example/img/800px-image.gif")) 800)))
+
+(define/contract (resize-image in new-width out)
+  (path? number? path? . -> . boolean?)
+  ;; Depend on ImageMagick
+  (define (magick-resize in w out)
+    (system (format "magick \"~a\" -geometry ~a \"~a\"" (path->string in) w (path->string out))))
+  (define resize magick-resize)
```

Comment:
I'm not sure why this? Couldn't you simply rename `magick-resize` to `resize` (if you prefer that) in the first place?